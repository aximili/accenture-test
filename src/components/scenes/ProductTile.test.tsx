import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTile } from '.';

test('Renders Product not on sale', () => {
    const product = {
        index: 0,
        isSale: false,
        priceText: "$49.99",
        productImage: "noimage.jpg",
        productName: "Pure Blonde Crate",
        type: "Beer"
    };

    const dom = renderer.create(
        <ProductTile product={product} />
    ).toJSON();

    expect(dom).toMatchSnapshot();
});

test('Renders Product on sale', () => {
    const product = {
        index: 0,
        isSale: true,
        priceText: "$49.99",
        productImage: "noimage.jpg",
        productName: "Pure Blonde Crate",
        type: "Beer"
    };

    const dom = renderer.create(
        <ProductTile product={product} />
    ).toJSON();

    expect(dom).toMatchSnapshot();
});