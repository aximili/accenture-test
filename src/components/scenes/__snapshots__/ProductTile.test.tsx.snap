// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Product not on sale 1`] = `
<div
  className="MuiPaper-root MuiCard-root product-tile MuiPaper-elevation3 MuiPaper-rounded"
>
  <div
    className="image-wrapper"
  >
    <img
      alt="Pure Blonde Crate"
      onError={[Function]}
      src="/images/noimage.jpg"
    />
  </div>
  <div
    className="product-details"
  >
    <div>
      Pure Blonde Crate
    </div>
    <div
      className="price"
    >
      $49.99
    </div>
  </div>
</div>
`;

exports[`Renders Product on sale 1`] = `
<div
  className="MuiPaper-root MuiCard-root product-tile MuiPaper-elevation3 MuiPaper-rounded"
>
  <div
    className="sale-banner"
  >
    Sale
  </div>
  <div
    className="image-wrapper"
  >
    <img
      alt="Pure Blonde Crate"
      onError={[Function]}
      src="/images/noimage.jpg"
    />
  </div>
  <div
    className="product-details"
  >
    <div>
      Pure Blonde Crate
    </div>
    <div
      className="price sale"
    >
      $49.99
    </div>
  </div>
</div>
`;
